saleor/payment/forms.py

```diff
@@ -0,0 +1,83 @@
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import pgettext_lazy
+
+from . import ChargeStatus, TransactionType, get_provider
+from .models import Payment
+from .utils import gateway_authorize, gateway_capture, gateway_refund
+
+
+def get_form_for_payment(payment):
```

Comment:
Is this debug or is it supposed to stay like that?

Reply:
It's supposed to stay like that. Docs will be expanded once I'll have a more steady version of the old checkout. Preferably Monday

Reply:
If it's supposed to stay that way, I'm fully against it. Here is my idea on the matter:

We could reuse the retrieved provider to retrieve the form instead of deciding which form to use from the payment variant name through a list of "hard-coded" cases.
https://github.com/mirumee/saleor/blob/a4f77a9d22216905738c6442867871f0b48ea1fb/saleor/order/views.py#L103-L106

Instead of getting the form from a payment method, I would suggest doing **`provider.get_form_class`** that would return a pointer to `saleor.payment.providers.braintree.forms.BraintreePaymentForm`. So the design pattern would be having the provider form, widgets, etc. in **`forms.py`**:
```
payment/
├── forms.py
├── __init__.py
├── providers
    ├── braintree
    │   ├── forms.py
    │   └── __init__.py
    ├── dummy
    │   ├── forms.py
    │   └── __init__.py
    └── __init__.py
```

Reply:
Yup, that's better in a long term